### PR TITLE
Add labelComponents option to form control has label

### PR DESCRIPTION
**Pull request by gavmck**
We have custom label components in our app which means the `form-control-has-label` rule is always triggered. 

I've leant on the options setup for `label-has-for` and implemented a similar setup where you can pass an array of custom components to be considered as `label` elements by the rule for validation.

In the app template:

```
<CustomLabel for="myInput">Label</CustomLabel>
<input type="text" id="myInput" />
```

Pass the label components as part of the ruleset:

```
"vuejs-accessibility/form-control-has-label": [
      "error",
      {
        "labelComponents": ["CustomLabel"],
      }
    ]
```

### CODE REVIEW

File: src/rules/form-control-has-label.ts [hunk 1]
```diff
@@ -42,8 +49,10 @@ const rule: Rule.RuleModule = {
   create(context) {
```
Comment (vhoyer) [inline src/rules/form-control-has-label.ts:49]: hey, can you add the `scheme` about the option you just added? I think we got an example in `label-has-for`, I don't know if it's super necessary, but I think the rules that have option have those

File: src/rules/form-control-has-label.ts [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
 } from "../utils";
 
 function isLabelElement(
   node:
     | AST.VElement
     | AST.VDocumentFragment
     | AST.VText
-    | AST.VExpressionContainer
+    | AST.VExpressionContainer,
+  options: FormControlHasLabelOptions
```
Comment (vhoyer) [inline src/rules/form-control-has-label.ts:24]: (optional)
```suggestion
  { labelComponents = [] }: FormControlHasLabelOptions
```

then the line 26 can be like:
```js
  const labelComponents = labelComponents.map(makeKebabCase).concat("label");
```

File: src/rules/form-control-has-label.ts [hunk 3]
```diff
@@ -1,31 +1,38 @@
...
 ) {
-  return node.type === "VElement" && getElementType(node) === "label";
+  const labelComponents = (options.labelComponents || []).map(makeKebabCase).concat("label");
+  return node.type === "VElement" && labelComponents.includes(getElementType(node));
```
Comment (vhoyer) [inline src/rules/form-control-has-label.ts:27]: (resolvable) I think it's kinda of exposing implementation details having to use the `makeKebabCase` to compare stuff using `getElementType` I guess having a proper `isMatchingElement` or something could maybe better, then we wouldn't have to remember that we need to normalize stuff using `makeKebabCase` before comparing using `getElementType`, do you guys understand?

could be something like:

```js
export default function isMatchingElement(node, searchArray) {
  const elementType = getElementType(node);
  
  return searchArray.some(item => {
    return makeKebabCase(item) === elementType;
  });
}
```
But you know, just a suggestion.

File: src/rules/form-control-has-label.ts (hunk 1 above)
Comment (gavmck) [reply src/rules/form-control-has-label.ts:49]: Assuming you mean the `schema`? No worries, updated.

File: src/rules/form-control-has-label.ts (hunk 2 above)
Comment (gavmck) [reply src/rules/form-control-has-label.ts]: Tweaked slightly to prevent overwriting the variable

File: src/rules/form-control-has-label.ts (hunk 3 above)
Comment (gavmck) [reply src/rules/form-control-has-label.ts]: Yeah that seems cleaner, done

**Review by vhoyer — COMMENTED** [review]
Correct me if I'm wrong, but i don't see tests for this feature on the diff, could you add them?

Another thing, I think we will need to ask the maintainer about this, but I think you don't need to update te version on this PR 🤔

**Comment by gavmck** [comment]
> Correct me if I'm wrong, but i don't see tests for this feature on the diff, could you add them?
> 
> Another thing, I think we will need to ask the maintainer about this, but I think you don't need to update te version on this PR 🤔

Ah I missed the tests folder! Updated.

**Review by vhoyer — APPROVED** [review]
Seems good to me, wdyt @kddnewton ?

**Comment by gavmck** [comment]
Any update on this? Still throwing warnings all over the shop in our app.

**Comment by gavmck** [comment]
Fixed the conflicts, reckon this can get merged in?

**Comment by shaunkrd** [comment]
Nice addition 👍🏼 Would be great to see this added 🙏

**Pull request merged**